com.ibm.streamsx.plumbing/com.ibm.streamsx.plumbing.synchronize/Synchronize/Synchronize.xml

```diff
@@ -0,0 +1,130 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--                                                                        -->
+<!-- Copyright (C) 2013, 2016 International Business Machines Corporation.  -->
+<!-- All Rights Reserved.                                                   -->
+<!--                                                                        -->
+<operatorModel
+  xmlns="http://www.ibm.com/xmlns/prod/streams/spl/operator" 
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xmlns:cmn="http://www.ibm.com/xmlns/prod/streams/spl/common"
+  xsi:schemaLocation="http://www.ibm.com/xmlns/prod/streams/spl/operator operatorModel.xsd">
+  <cppOperatorModel>
+
+    <context>
+      <description>
+
+The Synchronize operator synchronizes the flow of tuples across multiple streams according to timestamps in the tuples.  It can be configured with one or more input ports, and must have the same number of output ports.  Optionally, it can pace tuples faster or slower than they were produced, according to their timestamps.
+
+The Synchronize operator does not alter tuples that pass through it. All tuples are forwarded as received to the output port corresponding to their input port.
+
+The Synchronize operator synchronizes streams by receiving tuples in all of its input ports and forwarding them downstream in order according to their timestamps, temporarily holding tuples in faster streams until slower streams catch up.  Optionally, the operator can pace tuples by delaying them according to the differences between their timestamps, according to a time-scaling factor. 
```

Comment:
Yes, the operator needs to receive a tuple on each input port before it can determine which one has the smallest timestamp value. So, in your example, it would forward B(1), B(2), and B(3) while holding A(100), and then continue to hold A(100) until it receives B(5000). Then it would forward A(100) and continue to hold B(5000) until another tuple arrives on port A. Then it would decide whether to forward the A tuple or the B tuple next.

If the time scaling factor is not zero, the operator will wait a while before forwarding the next tuple. The delay is the difference between the timestamp values of the next and previous tuples (across all streams), divided by the scaling factor. In your example, the delay between B(1), B(2), and B(3) will be really small, and the delay before B(5000) will be really large.

I'll add this to the description.